src/nc/core/likec/Simplifier.cpp

```diff
@@ -400,7 +416,45 @@ std::unique_ptr<MemberAccessOperator> Simplifier::simplify(std::unique_ptr<Membe
 
 std::unique_ptr<Expression> Simplifier::simplify(std::unique_ptr<Typecast> node) {
     node->operand() = simplify(std::move(node->operand()));
-
+    /*
+     * reinterpret_cast<int16_t>(*reinterpret_cast<uint16_t*>(&edi2))
+     * will become *reinterpret_cast<int16_t*>(&edi2)
+     */
+    if(node->operand()->is<UnaryOperator>()
+                                          && node->type()->isInteger() &&
+            node->operand()->as<UnaryOperator>()->operatorKind() == UnaryOperator::DEREFERENCE) {
```

Comment:
I would put `node->operand()` checks next to each other: related things should be closer than non-related things.